### ISSUE TITLE
Tool does not work on ubuntu 18.04 with GA kernel

### ISSUE THREAD

**Issue by VMinute** [Sysinternals/ProcMon-for-Linux#10]
When running on ubuntu 18.04 the tool doesn't show any information. I can see the main screen, but number of event is stuck at 0, even after a few minutes spent running other apps etc.

Minimum requirements list just ubuntu 18.04 but on my Ubuntu 18.04 machine the tool shows no information and when I quit I get this message on the terminal:
```
HINT: bpf_get_stack missing (added in Linux 4.18).
```

My PC is actually running 4.15 that is the default kernel release for ubuntu 18.04:
```
# lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.4 LTS
Release:	18.04
Codename:	bionic
# uname -a
Linux valterP52S 4.15.0-111-generic #112-Ubuntu SMP Thu Jul 9 20:32:34 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```

Here's the full output from the tool:
```
# sudo procmon
bpf: Failed to load program: Invalid argument
0: (bf) r8 = r1
1: (b7) r9 = 0
2: (63) *(u32 *)(r10 -8) = r9
3: (18) r1 = 0xffff956918a24200
5: (bf) r2 = r10
6: (07) r2 += -8
7: (85) call bpf_map_lookup_elem#1
8: (55) if r0 != 0x0 goto pc+1
 R0=inv0 R8=ctx(id=0,off=0,imm=0) R9=inv0 R10=fp0
9: (05) goto pc+6
16: (85) call bpf_get_current_pid_tgid#14
17: (bf) r7 = r0
18: (7b) *(u64 *)(r10 -16) = r7
19: (b7) r1 = 0
20: (63) *(u32 *)(r10 -8) = r1
21: (18) r1 = 0xffff956856ac7000
23: (bf) r2 = r10
24: (07) r2 += -8
25: (85) call bpf_map_lookup_elem#1
26: (55) if r0 != 0x0 goto pc+1
 R0=inv0 R7=inv(id=0) R8=ctx(id=0,off=0,imm=0) R9=inv0 R10=fp0
27: (05) goto pc+164
192: (79) r1 = *(u64 *)(r8 +8)
193: (63) *(u32 *)(r10 -20) = r1
194: (18) r1 = 0xffff956856ac5a00
196: (bf) r2 = r10
197: (07) r2 += -20
198: (85) call bpf_map_lookup_elem#1
199: (18) r9 = 0xffffffff
201: (15) if r0 == 0x0 goto pc+101
 R0=map_value(id=0,off=0,ks=4,vs=728,imm=0) R7=inv(id=0) R8=ctx(id=0,off=0,imm=0) R9=inv4294967295 R10=fp0
202: (7b) *(u64 *)(r10 -32) = r0
203: (b7) r1 = 0
204: (63) *(u32 *)(r10 -24) = r1
205: (18) r1 = 0xffff956918a24800
207: (bf) r2 = r10
208: (07) r2 += -24
209: (85) call bpf_map_lookup_elem#1
210: (bf) r7 = r0
211: (18) r9 = 0xffffffff
213: (15) if r7 == 0x0 goto pc+89
 R0=map_value(id=0,off=0,ks=4,vs=704,imm=0) R7=map_value(id=0,off=0,ks=4,vs=704,imm=0) R8=ctx(id=0,off=0,imm=0) R9=inv4294967295 R10=fp0 fp-32=map_value
214: (79) r1 = *(u64 *)(r10 -16)
215: (63) *(u32 *)(r7 +0) = r1
 R0=map_value(id=0,off=0,ks=4,vs=704,imm=0) R1=inv(id=0) R7=map_value(id=0,off=0,ks=4,vs=704,imm=0) R8=ctx(id=0,off=0,imm=0) R9=inv4294967295 R10=fp0 fp-32=map_value
216: (61) r1 = *(u32 *)(r10 -20)
217: (63) *(u32 *)(r7 +4) = r1
 R0=map_value(id=0,off=0,ks=4,vs=704,imm=0) R1=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R7=map_value(id=0,off=0,ks=4,vs=704,imm=0) R8=ctx(id=0,off=0,imm=0) R9=inv4294967295 R10=fp0 fp-32=map_value
218: (bf) r2 = r7
219: (07) r2 += 24
220: (bf) r1 = r8
221: (b7) r3 = 256
222: (b7) r4 = 256
223: (85) call unknown#67
invalid func unknown#67

HINT: bpf_get_stack missing (added in Linux 4.18).
```

**Comment by skyrider223** [comment]
Hello @VMinute,

Maybe your kernel is too old ? 
Linux valterP52S **4.15**.(...) < HINT: bpf_get_stack missing (added in Linux **4.18**)

Cheers,

**Comment by VMinute** [comment]
For sure my kernel is too old, I hoped it was clear in the message :)
But this is the standard kernel for ubuntu 18.04. You may update it (I will probably try as soon as I'll not have too many urgent things to finish), but that is not the standard.
So having "ubuntu 18.04" as a requirement is misleading.
Maybe the ubuntu package itself could check kernel version and give a warning?

**Comment by MarioHewardt** [comment]
You are correct, apologies for that. One particular capability of procmon is collection of stack traces and this does indeed require 4.19 (most other capabilities should work on 4.15). One approach here would be to have procmon check the kernel version and only enable the capabilities that the kernel supports (to some degree at least, not sure if we want to get too granular).

**Comment by tigros** [comment]
i had this problem too so discovered ukuu and put 5.7.1 but same symptoms then tried 4.20.17 and it works. i just thought 5.7.1 issue might be of interest.

**Comment by WSLUser** [comment]
I would consider the kernel version you get from `apt upgrade` to be standard for Ubuntu. Doesn't matter it's not the initial one when Ubuntu 18.04 releases. I would consider GA to be the latest available from the Ubuntu repositories for that distro version.

**Comment by VMinute** [comment]
I regularly do apt upgrade of my system, but kernel version didn't change.
You have to install the HWE kernel to update it, but I would not consider that part of a "standard" update procedure.
Specifying Ubuntu 18.04.4 or Ubuntu 18.04 with kernel v XXX would make the documentation better, IMVHO.

**Comment by cmaritan** [comment]
Hello everyone, could it be useful to:

1. Update docs

2. Add a check on kernel version at runtime using "sys/utsname.h" and stop execution if does not match requirements (ver. < 4.18)

3. Add a check on kernel version in CMake. Maybe only a warning (It compiles but will not work in the system where it has been built).

I'm a beginner but I can try to submit PR if It's ok for @jahabibi , @TETRA2000 and @josalem (btw: are you the maintainers of the project?).

Bye.

**Comment by MarioHewardt** [comment]
Thanks for the feedback @cmaritan. We're doing some work in terms of making the capabilities of the different kernels more user friendly in procmon. Stay tuned on that. For the time being though, if you want to create a PR for doc changes that would be much appreciated. Please add @MarioHewardt to the PR (@MarioHewardt, @jahabibi, @josalem are the maintainers).

**Comment by MarioHewardt** [comment]
This should be fixed with procmon 1.1
